src/Adapter/Customer/Group/Repository/GroupRepository.php

```diff
@@ -49,4 +52,17 @@ public function assertGroupExists(GroupId $groupId): void
             GroupNotFoundException::class
         );
     }
+
+    public function create(array $localizedNames, DecimalNumber $reduction, bool $priceDisplayMethod, bool $showPrices): CustomerGroup
+    {
+        $customerGroup = new CustomerGroup();
+        $customerGroup->name = $localizedNames;
+        $customerGroup->reduction = (string) $reduction;
+        $customerGroup->price_display_method = (int) $priceDisplayMethod;
+        $customerGroup->show_prices = $showPrices;
+
+        $this->addObjectModel($customerGroup, CannotAddGroupException::class);
```

Comment:
I think you should build it in the command no?

Reply:
no its all good, In handler he will have to call $repository->create({$commandArgs...}). So we don't deal with object model directly in handler and we can wrap this all into some interface when ADR's about repos/entities interfaces are finished.